src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -52,6 +48,46 @@ interface IAdaptationSwitchingInfos  {
   };
 }
 
+/**
+ * Detect if the accessibility given defines an adaptation for the visually
+ * impaired.
+ * Based on DVB Document A168 (DVB-DASH).
+ * @param {Object} accessibility
+ * @returns {Boolean}
+ */
+function isVisuallyImpaired(
+  accessibility: { schemeIdUri? : string; value? : string }
+) : boolean {
+  if (!accessibility) {
+    return false;
+  }
+
+  return (
+    accessibility.schemeIdUri === "urn:tva:metadata:cs:AudioPurposeCS:2007" &&
+    accessibility.value === "1"
+  );
+}
+
+/**
+ * Detect if the accessibility given defines an adaptation for the hard of
+ * hearing.
+ * Based on DVB Document A168 (DVB-DASH).
+ * @param {Object} accessibility
+ * @returns {Boolean}
+ */
+function isHardOfHearing(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 58. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=431&issues=AWfg0Pzz1_eUtAp8jYhy&open=AWfg0Pzz1_eUtAp8jYhy)

[ISSUE_KEY]: AWfg0Pzz1_eUtAp8jYhy